src/server/routes/signIn.ts

```diff
@@ -283,6 +304,268 @@ const oktaSignInControllerErrorHandler = (error: unknown): SignInError => {
 	};
 };
 
+/**
+ * @name oktaIdxApiSignInController
+ * @description Start the Okta IDX flow to attempt to sign the user in with a password
+ *
+ * @param {Request} req - Express request object
+ * @param {ResponseWithRequestState} res - Express response object
+ * @returns {Promise<void>}
+ */
+const oktaIdxApiSignInController = async ({
+	req,
+	res,
+}: {
+	req: Request;
+	res: ResponseWithRequestState;
+}) => {
+	// TODO: remove when the useIdxSignIn feature flag is removed
+	// placeholder warning message
+	logger.warn(
+		'IDX API password authentication flow is not fully implemented yet',
+	);
+
+	// get the email and password from the request body
+	const { email = '', password = '' } = req.body;
+
+	try {
+		// First we want to check the user status in Okta
+		// to see if they are in the ACTIVE state
+		// if they are not, we will not allow them to sign in
+		const user = await getUser(email, req.ip).catch((error) => {
+			// handle any getUser errors here instead of the outer catch block
+			if (isOktaError(error)) {
+				// convert the user not found error to generic authentication error to outer catch block
+				if (error.status === 404 && error.code === 'E0000007') {
```

Comment:
https://github.com/guardian/gateway/blob/c6c98b49bb86ae965b252e58068009eb8c0527b0/src/server/models/okta/Error.ts#L33-L41

Yeah these were set up a long time ago, they correspond directly to errors from Okta. Ideally rather than just being a `string` type, theres a possible refactor here to change them to a literal type, or a user readable type using this map.